packages/remix-react/components.tsx

```diff
@@ -1248,16 +1248,29 @@ export interface SubmitFunction {
  *
  * @see https://remix.run/api/remix#usesubmit
  */
-export function useSubmit(): SubmitFunction {
-  return useSubmitImpl();
+export function useSubmit(options?: SubmitOptions): SubmitFunction {
+  let key = undefined; // no key for global submit
+  return useSubmitImpl(key, options);
 }
 
-let defaultMethod = "get";
-let defaultEncType = "application/x-www-form-urlencoded";
-
-export function useSubmitImpl(key?: string): SubmitFunction {
+const DEFAULT_METHOD = "get";
+const DEFAULT_ENC_TYPE = "application/x-www-form-urlencoded";
+
+// IDEA: remove the export here?
+export function useSubmitImpl(
+  key?: string,
+  {
+    method: submitMethod = DEFAULT_METHOD,
+    action: submitAction,
+    encType: submitEncType = DEFAULT_ENC_TYPE,
+    replace,
+  }: SubmitOptions = {}
+): SubmitFunction {
   let navigate = useNavigate();
-  let defaultAction = useFormAction();
+  let defaultFormAction = useFormAction();
+  let defaultAction = submitAction || defaultFormAction;
+  let defaultMethod = submitMethod || DEFAULT_METHOD;
+  let defaultEncType = submitEncType || DEFAULT_ENC_TYPE;
```

Comment:
By putting these calculations all here we can avoid having to pass `defaultFormAction` into the dependency array which is what fixes my original issue.